controllers/storagecluster/placement.go

```diff
@@ -22,10 +22,20 @@ func getPlacement(sc *ocsv1.StorageCluster, component string) rookCephv1.Placeme
 		(&in).DeepCopyInto(&placement)
 		// label rook_file_system is added to the mds pod using rook operator
 		if component == "mds" {
-			placement.PodAntiAffinity = &corev1.PodAntiAffinity{
-				PreferredDuringSchedulingIgnoredDuringExecution: []corev1.WeightedPodAffinityTerm{
-					defaults.GetMdsWeightedPodAffinityTerm(100, generateNameForCephFilesystem(sc)),
-				},
+			// if active MDS number is more than 1 then Preferred and if it is 1 then Required pod anti-affinity is set
+			mdsWeightedPodAffinity := defaults.GetMdsWeightedPodAffinityTerm(100, generateNameForCephFilesystem(sc))
+			if sc.Spec.ManagedResources.CephFilesystems.ActiveMetadataServers > 1 {
```

Comment:
@malayparida2000 can we also list the total nodes and see if the storage nodes are more than a number of pods?

Reply:
@parth-gr I didn't do that as this fix needs to be backported till 4.15 & I don't want to add client list operations in backports. We can implement the get node & decision based on that may be in 4.19.

Reply:
@malayparida2000 the fix for using filesystem name as key for antiaffinity is even not backported,

So can have this in 4.18,

And for backports we can have seperate PR in 4.17

Reply:
Went through the code a bit. To list & get odf labeled nodes We will need a change in the getPlacement() funcs signature as it would now need the client. This would require a lot of changes in the codebase especially the unit tests. Considering tomorrow is the RC I don't think it's achievable by tomorrow.